sklearn/decomposition/dict_learning.py

```diff
@@ -669,8 +670,9 @@ def dict_learning_online(X, n_components=2, alpha=1, n_iter=100,
                 print ("Iteration % 3i (elapsed time: % 3is, % 4.1fmn)"
                        % (ii, dt, dt / 60))
 
+        # Setting n_jobs = 1 to avoid creating workers at each iteration
         this_code = sparse_encode(this_X, dictionary.T, algorithm=method,
-                                  alpha=alpha, n_jobs=n_jobs).T
+                                  alpha=alpha, n_jobs=1).T
```

Comment:
There is a gain in the transform / sparse_encode function,once the dictionary has been computed. However, during the alternative process of online dictionary learning (sparse_encode -> _update_dict), using n_jobs > 1 actually slows down the process (see #4769). This is a joblib inherent issue, as each iteration requires to set up new workers